R/style_guides.R

```diff
@@ -186,3 +193,45 @@ character_to_ordered <- function(x, levels, name = substitute(x)) {
   factor(x, levels = levels, ordered = TRUE)
 }
 
+#' Specify spacing around math tokens
+#'
+#' Helper function to create the input for the argument `math_token_spacing`  in
+#' [tidyverse_style()].
+#' @inheritParams style_space_around_math_token
+#' @examples
+#' style_text(
+#'   "1+1   -3",
+#'   math_token_spacing = specify_math_token_spacing(zero = "'+'"),
+#'   strict = FALSE
+#' )
+#' style_text(
+#'   "1+1   -3",
+#'   math_token_spacing = specify_math_token_spacing(zero = "'+'"),
+#'   strict = TRUE
+#' )
+#' @export
+specify_math_token_spacing <-
```

Comment:
Maybe `default_math_token_spacing`?

Reply:
Not sure whether I understand you correctly. What does it have to do with a default? Everything will be specified unambiguously. We can also drop the `specify` from the name in order to make it sound a bit more simplistic.

Reply:
Oh, I see. I only saw the use of the function as a default argument. Maybe implement two functions then, `math_token_spacing()` and `default_math_token_spacing()`, and have the latter forward to the former?

Reply:
Ok, we can do that.

Reply:
I am not sure whether we need the forwarder. I.e. if we only have one style guide, then that might be fine, but other style guides may have other defaults. So maybe call it `tidyverse_math_token_spacing()`?

Reply:
We cannot use a function that has the same name as an argument (`math_token_spacing()`):
```r
tidyverse_style <- function(scope = "tokens",
                            strict = TRUE,
                            indent_by = 2,
                            start_comments_with_one_space = FALSE,
                            math_token_spacing = math_token_spacing()) {
# [...]
```
Because we get 
```r
#> Error: promise already under evaluation: recursive default argument reference or earlier
#> problems?
```

I suggest to revert to `specify_math_token_spacing()`, since that is also a verb and functions are supposed to be verbs according to the tidyverse style guide  I think.

Reply:
@krlmlr ok? We can keep `tidyverse_math_token_spacing()` that forwards to `specify_math_token_spaching()`.